go-controller/pkg/kubevirt/pod.go

```diff
@@ -479,3 +479,37 @@ func DiscoverLiveMigrationStatus(client *factory.WatchFactory, pod *corev1.Pod)
 	}
 	return &status, nil
 }
+
+func ReconcileIPv4DefaultGatewayAfterLiveMigration(watchFactory *factory.WatchFactory, netInfo util.NetInfo, liveMigrationStatus *LiveMigrationStatus, interfaceName string) error {
+	if liveMigrationStatus.State != LiveMigrationTargetDomainReady {
+		return nil
+	}
+
+	targetNode, err := watchFactory.GetNode(liveMigrationStatus.TargetPod.Spec.NodeName)
+	if err != nil {
+		return err
+	}
+
+	lrpJoinAddress, err := util.ParseNodeGatewayRouterJoinNetwork(targetNode, netInfo.GetNetworkName())
+	if err != nil {
+		return err
+	}
+
+	lrpJoinIPv4, _, err := net.ParseCIDR(lrpJoinAddress.IPv4)
+	if err != nil {
+		return err
+	}
+
+	lrpMAC := util.IPAddrToHWAddr(lrpJoinIPv4)
+	for _, subnet := range netInfo.Subnets() {
+		gwIP := util.GetNodeGatewayIfAddr(subnet.CIDR).IP.To4()
+		if gwIP == nil {
+			continue
+		}
```

Comment:
When does this happen? When the subnet is IPv6? What happens if the subnet is IPv6?

Reply:
If subnet is IPv6 we don't have to reconcile the gateway with this mechanism, we have other PRs for IPv6 doing similar stuff, but they have less priority than this.

Reply:
But that doesn't answer my question

Reply:
This happends after VM live migration success, the subne is ipv6 if cluster is dual stack or single ipv6 and the NAD has ipv6 subnet at the NAD, if the subnet is ipv6 the gateway is configureding using router advetisments from gateway router LRP,

Reply:
I can't understand what you are saying.
Are you trying to say that `gwIP == nil` when `util.GetNodeGatewayIfAddr(subnet.CIDR).IP` is an IPv6 address? And thus we skip it because this only intends to handle IPv4?

Reply:
The golang net.IP.To4 returns nil if it's not a IPv4 -> https://pkg.go.dev/net#IP.To4

So the idea is to skip this if it's not IPv4.